api/envoy/extensions/http/custom_response/redirect_policy/v3/redirect_policy.proto

```diff
@@ -26,28 +27,78 @@ option (xds.annotations.v3.file_status).work_in_progress = true;
 message RedirectPolicy {
-  // [#comment: TODO(pradeepcrao): Add the ability to specify the full uri, or just host or
-  // path rewrite for the redirection in the same vein as
-  // config.route.v3.RedirectAction]
-  // The host that will serve the custom response.
-  //
-  // Example:
-  //
-  // .. code-block:: yaml
-  //
-  //    uri: https://www.mydomain.com
-  //
-  string host = 1 [(validate.rules).string = {min_len: 1}];
-
-  // The path for the custom response.
-  //
-  // Example:
-  //
-  //  .. code-block:: yaml
-  //
-  //    path: /path/to/503_response.txt
-  //
-  string path = 2 [(validate.rules).string = {min_len: 1}];
+  // [#next-free-field: 8]
+  message RedirectAction {
+    // When the scheme redirection take place, the following rules apply:
+    //  1. If the source URI scheme is ``http`` and the port is explicitly
+    //     set to ``:80``, the port will be removed after the redirection
+    //  2. If the source URI scheme is ``https`` and the port is explicitly
+    //     set to ``:443``, the port will be removed after the redirection
+    oneof scheme_rewrite_specifier {
+      // The scheme portion of the URL will be swapped with "https".
+      bool https_redirect = 1;
+
+      // The scheme portion of the URL will be swapped with this value.
+      string scheme_redirect = 2;
+    }
+
+    // The host portion of the URL will be swapped with this value.
+    // Note that at least one of host_redirect and path_redirect has to be
+    // specified.
+    string host_redirect = 3
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
+
+    // The port value of the URL will be swapped with this value.
+    google.protobuf.UInt32Value port_redirect = 4;
+
+    // The path portion of the URL will be swapped with this value.
+    // Please note that query string in path_redirect will override the
+    // request's query string and will not be stripped.
+    //
+    // For example, let's say we have the following routes:
+    //
+    // - match: { path: "/old-path-1" }
+    //   redirect: { path_redirect: "/new-path-1" }
+    // - match: { path: "/old-path-2" }
+    //   redirect: { path_redirect: "/new-path-2", strip-query: "true" }
+    // - match: { path: "/old-path-3" }
+    //   redirect: { path_redirect: "/new-path-3?foo=1", strip_query: "true" }
+    //
+    // 1. if request uri is "/old-path-1?bar=1", users will be redirected to "/new-path-1?bar=1"
+    // 2. if request uri is "/old-path-2?bar=1", users will be redirected to "/new-path-2"
+    // 3. if request uri is "/old-path-3?bar=1", users will be redirected to "/new-path-3?foo=1"
+    //
+    // Note that at least one of host_redirect and path_redirect has to be
+    // specified.
+    string path_redirect = 5
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
+
+    // Indicates that during redirection, the query portion of the URL will
+    // be removed. Default value is false.
```

Comment:
The description only matches `strip_query`. I suggest moving the comments next to each of the fields.